.github/actions/release-pr/action.yml

```diff
@@ -0,0 +1,249 @@
+              exit 1
+            fi
+          fi
+
+          echo "Version ${{ inputs.version }} is valid."
+
+          # Construct BRANCH_VERSION as MAJOR.MINOR + EXTRA (retain pre-release and build metadata)
+          BRANCH_VERSION="${MAJOR}.${MINOR}${EXTRA}"
+          echo "Branch version: $BRANCH_VERSION"
+
+          if [[ "${{ inputs.release-type }}" == "major" ]]; then
+            echo "PRIOR_VERSION=$(( MAJOR - 1 ))" | tee -a $GITHUB_ENV
+          elif [[ "${{ inputs.release-type }}" == "minor" ]]; then
+            echo "PRIOR_VERSION=${MAJOR}" | tee -a $GITHUB_ENV
+          fi
+        else
+          echo "Version ${{ inputs.version }} is not valid SemVer. Aborting..."
+          exit 1
+        fi
+
+        echo "BRANCH_VERSION=$BRANCH_VERSION" | tee -a $GITHUB_ENV
+        echo "CRATE_VERSION=${{ inputs.version }}" | tee -a $GITHUB_ENV
+      shell: bash
+
+    # Always checks out an existing release branch (if existent) as the base for the release PR. Maintainers are expected to
+    # cherry-pick the desired changes from the default branch and push to the temporary `release-pr` branch before merging the PR.
+    #
+    # The `release/<tag-prefix>-v<version>` branch always removes the patch number from `<version>` as we don't want to change
+    # the release branch name for patch versions.
+    # However, when we make a version bump, we include the full SemVer input in `Cargo.toml` and the release tag for clarity
+    - name: Create or checkout base branch
+      run: |
+        if [[ -n "${{ inputs.tag-prefix }}" ]]; then
+          TAG_PREFIX="${{ inputs.tag-prefix }}-"
+        fi
+
+        BASE_BRANCH="release/${TAG_PREFIX}v${{ env.BRANCH_VERSION }}"
+        echo $BASE_BRANCH
+
+        # If major release, get latest minor release from prior major version
+        # If minor release, get latest minor release from given major version
+        if [[ "${{ inputs.release-type }}" == "major" || "${{ inputs.release-type }}" == "minor" ]]; then
+          set +o pipefail
+          PRIOR_RELEASE=$(git branch -r | grep -E "origin/release/${TAG_PREFIX}v${{ env.PRIOR_VERSION }}\.[0-9]+" | sed 's/origin\///' | sort -V | tail -n 1)
+          set -o pipefail
+          echo "Prior release: $PRIOR_RELEASE"
+          # If a prior release is found, check it out as the source of the new release branch (which is the PR base)
+          # Otherwise, use the default branch (e.g. if this is the first release).
+          if [[ -n "$PRIOR_RELEASE" ]]; then
+            git checkout $PRIOR_RELEASE
+          fi
+          git checkout -b $BASE_BRANCH
+          git push origin $BASE_BRANCH
+        else
+          git checkout $BASE_BRANCH
+        fi
+
+        echo "BASE_BRANCH=$BASE_BRANCH" | tee -a $GITHUB_ENV
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      shell: bash
+
+    # Include the full patch version in the PR branch name if a patch, otherwise call it `release-pr` for a major or minor release
```

Comment:
It might also be interesting to have different names for release branches as some repositories might have multiple releases of major/minor that could be triggered in parallel.

If that's too bothersome we'll just constrain ourselves to keep it to one at a time.